plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -89,29 +113,45 @@ private static boolean hasUserOptedIn() {
    * opting out. (Therefore, this method essentially returns true if the opt-in dialog was ever
    * presented before.)
    */
-  private static boolean hasUserRegisteredOptInStatus() {
-    return getBooleanPreference(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
+  private boolean hasUserRegisteredOptInStatus() {
+    return preferences.getBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
   }
 
-  static void registerOptInStatus(boolean optedIn) {
-    IEclipsePreferences prefs = ConfigurationScope.INSTANCE.getNode(PREFERENCES_PLUGIN_ID);
-    if (prefs != null) {
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN, optedIn);
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, true);
-      flushPreferences(prefs);
-    }
+  void registerOptInStatus(boolean optedIn) {
```

Comment:
Ah, I get your point. Currently, the OPT_IN_REGISTERED cannot be conditioned on OPT_IN itself. The only way we can get rid of OPT_IN_REGISTERED is to use a three-valued setting like I said: OPTED_IN, OPTED_OUT, and UNDEF. I admit that employing a three-valued setting is conceptually ideal, so I will create an issue for this and fix it in another PR.